libraries/WiFi/src/WiFiGeneric.cpp

```diff
@@ -1608,6 +1632,37 @@ int WiFiGenericClass::hostByName(const char* aHostname, IPAddress& aResult)
     return (uint32_t)aResult != 0;
 }
 
+/**
+ * Resolve the given hostname to an IP6 address.
+ * @param aHostname     Name to be resolved
+ * @param aResult       IPv6Address structure to store the returned IP address
+ * @return 1 if aHostname was successfully converted to an IP address,
+ *          else error code
+ */
+int WiFiGenericClass::hostByName6(const char* aHostname, ip_addr_t& aResult)
```

Comment:
I still think hostByName6 is not a good approach. See https://github.com/espressif/arduino-esp32/discussions/9009

Backwards compatibility should simply leave IPv6 off. Basic support of IPv6 wants hostByName() to support both 6 & 4 (whichever is relevant). Advanced IPv6 wants multiple address support.

Reply:
this will be changed. will be one version for all.